libcloudforensics/aws.py

```diff
@@ -0,0 +1,726 @@
+  # /configuration.html
+  return boto3.session.Session()
+
+
+class AwsAccount:
+  """Class representing an AWS account.
+
+  Attributes:
+    default_availability_zone (str): Default zone within the region to create
+    new resources in.
+  """
+  def __init__(self, default_availability_zone):
+    self.default_availability_zone = default_availability_zone
+    # The region is given by the zone minus the last letter
+    # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions
+    # -availability-zones.html#using-regions-availability-zones-describe
+    self.default_region = self.default_availability_zone[:-1]
+
+  def ClientApi(self, service, region=None):
+    """Create an AWS client object.
+
+    Attributes:
+      service (str): The AWS service to use.
+      region (str): The region is which to create new resources in.
+
+    Returns:
+      boto3.Session.Client: An AWS EC2 client object.
+    """
+    if region:
+      return CreateSession().client(
+          service_name=service, region_name=region
+      )
+    return CreateSession().client(
+        service_name=service, region_name=self.default_region
+    )
+
+  def ResourceApi(self, service, region=None):
+    """Create an AWS resource object.
+
+    Attributes:
+      service (str): The AWS service to use.
+      region (str): The region is which to create new resources in.
+
+    Returns:
+      boto3.Session.Resource: An AWS EC2 resource object.
+    """
+    if region:
+      return CreateSession().resource(
+          service_name=service, region_name=region
+      )
+    return CreateSession().resource(
+        service_name=service, region_name=self.default_region
+    )
+
+  def ListInstances(self, region=None, filters=None):
+    """List instances of an AWS account.
+
+    Attributes:
+      region (str): The region from which to list instances.
+      filters (list(dict)): Filters for the query.
+
+    Returns:
+      dict: Dictionary with name and metadata for each instance.
+
+    Example usage:
+      ListInstances(region='us-east-1', filters=[
+        dict(Name='instance-id', Values=['a-particular-instance-id'])
+      ])
+    """
+    if not filters:
+      filters = []
+
+    instances = dict()
+    have_all_tokens = False
+    next_token = None
+
+    while not have_all_tokens:
+      if next_token:
+        response = self.ClientApi(
+            EC2_SERVICE, region=region).describe_instances(
+                Filters=filters,
+                NextToken=next_token)
+      else:
+        response = self.ClientApi(
+            EC2_SERVICE, region=region).describe_instances(
+                Filters=filters)
+
+      for reservation in response["Reservations"]:
+        for instance in reservation["Instances"]:
+          instance_name = None
+          if instance.get('Tags'):
+            for tag in instance['Tags']:
+              instance_name = tag.get('Value') if tag.get('Key') == 'Name' \
+                else None
```

Comment:
`instance_name` is already set to None in L131. I would suggest the following change:

```python
if tag.get('Key') == 'Name':
  instance_name = tag.get('Value')
if instance_name is not None:
  break # or continue, see comment below.
```

Also, we may want to use `if instance_name:` without the `is not None:`. Using the latter means that you wanna treat empty* values (like `''`, `{}`, `[]`) in a different way than if the `Value` key is not set.

\* or values that evaluate to `False` when converted to a `bool`